synapse/handlers/message.py

```diff
@@ -319,8 +319,10 @@ async def get_joined_members(self, requester: Requester, room_id: str) -> dict:
                 room_id, user_id, allow_departed_users=True
             )
             if membership != Membership.JOIN:
-                raise NotImplementedError(
-                    "Getting joined members after leaving is not implemented"
+                raise SynapseError(
+                    code=403,
+                    errcode=Codes.FORBIDDEN,
+                    msg="Getting joined members after leaving is forbidden.",
```

Comment:
The fact that the membership isn't `Membership.JOIN` doesn't necessarily mean the user has left the room (implying they have first joined it at some point). It just means you're not currently in the room (e.g. maybe you've never joined it, or you've been kicked, etc). This is kind of on the previous error message for getting it wrong though.
I would write something like "You must be a current member of the room to request its joined members." or "Getting joined members while not being a current member of the room is forbidden."